lib/Dialect/Torch/IR/TorchOps.cpp

```diff
@@ -1175,6 +1175,37 @@ void AtenSizeOp::getCanonicalizationPatterns(RewritePatternSet &patterns,
   });
 }
 
+//===----------------------------------------------------------------------===//
+// AtenBroadcastToOp
+//===----------------------------------------------------------------------===//
+
+OpFoldResult AtenBroadcastToOp::fold(FoldAdaptor adaptor) {
+
+  SmallVector<int64_t> resultShape;
+  if (!matchPattern(this->getSize(), m_TorchListOfConstantInts(resultShape)))
+    return nullptr;
+
+  // May fold if any of the dims == -1, which means its value should be preserved.
+  auto isMinusOne = [&](int64_t &dimension) { return dimension == -1; };
+  if (llvm::none_of(resultShape, isMinusOne))
+    return nullptr;
+
+  auto selfType = this->getSelf().getType().dyn_cast<ValueTensorType>();
+  if (!selfType || !selfType.hasSizes())
+    return nullptr;
+
```

Comment:
Ah, you are removing the whole BroadcastToOp.

I was actually thinking to just update existing resultshape from -1 to the actual shape of the input in that dimension.
So if some broadcasting needs to happen, it can still happen. But we are guaranteed to not see a `-1` in the resultShape.
Now I realize that the resultShape are not an attribute on the op, but a different op, so this kind of thing cannot happen in the the ::fold method.

Anyway, does this change fix all the affected models? If yes, then its good enough and we don't need to do my more complicated idea.